mordant/src/commonMain/kotlin/com/github/ajalt/mordant/terminal/TerminalDetection.kt

```diff
@@ -14,11 +14,18 @@ internal object TerminalDetection {
     ): TerminalInfo {
         // intellij console is interactive, even though isatty returns false
         val ij = isIntellijRunActionConsole()
-        val inputInteractive = interactive ?: (ij || stdinInteractive())
-        val outputInteractive = interactive ?: (ij || stdoutInteractive())
+
+        // https://github.com/mintty/mintty/issues/482
+        val mintty = isMinttyTerminal()
+
+        val inputInteractive = interactive ?: (ij || mintty || stdinInteractive())
+        val outputInteractive = interactive ?: (ij || mintty || stdoutInteractive())
+
```

Comment:
This will cause ANSI codes to be output even if stdout is redirected to a file, which we want to avoid. So let's remove this part of the PR.

The linked issue mentions some native calls that can be used as a workaround, and we could add them to `Win32MppImpls` in a separate. I don't think that's necessary, though, since the issue also mentions that proper PTY support was added in 2019. It looks like Mordant can detect interactivity correctly on modern versions of mintty, so I would say we could omit all the workarounds and let folks on very old versions of mintty live without color.

Reply:
Actually, my initial code didn't include this `isMinttyTerminal()` check. But I had to add it as otherwise the `if (!interactive) return NONE` in `ansiLevel()` is taken and the `when (getTermProgram()) {` isn't even reached...

So...

> It looks like Mordant can detect interactivity correctly on modern versions of mintty

no, that did't work for me (using mintty 3.6.4).

> This will cause ANSI codes to be output even if stdout is redirected to a file, which we want to avoid.

True, but that's also already the case for `ij == true`, or?

Reply:
Also see https://github.com/mintty/mintty/wiki/Tips#inputoutput-interaction-with-alien-programs.

Reply:
> `when (getTermProgram()) {` isn't even reached...

Yes, that's the point; if stdout is redirected to a file, we don't want to output ansi codes.

> that's also already the case for ij == true

The intellij run console is a special case since it's not a normal terminal: you can't redirect it's output to a file, so we know it's always interactive.

> Also see https://github.com/mintty/mintty/wiki/Tips#inputoutput-interaction-with-alien-programs.

That seems to support the idea that PTY support should work on newer versions of mintty. Did you try setting and of the config/evvnars they mention?

Reply:
> Yes, that's the point; if stdout is redirected to a file, we don't want to output ansi codes.

Right. But *my* point is that `when (getTermProgram()) {` isn't reached even if output is *not* redirected to a file.

> The intellij run console is a special case

Ok, understood.

> That seems to support the idea that PTY support should work on newer versions of mintty.

You have to read closely: That's only the case when running mintty on Cygwin / MSYS2, i.e. if it uses the `cygwin1.dll` emulation layer. It's *not* the case when running mintty as an MinGW program, like it is done in the Git Bash that comes with Git for Windows. That's why Git for Windows itself has [this detection mechanism](https://github.com/git/git/blob/d4a83d07b8cc66d4afac2f33a8af729f2ba93bba/compat/winansi.c#L579-L585); we'd need something similar in Mordant.

> Did you try setting and of the config/evvnars they mention?

No. I'd like Mordant to work out of the box with a fresh installation of Git for Windows / mintty without any need to explicitly tell it that it's a color console; such an approach would not scale across users.

Reply:
> `when (getTermProgram()) {` isn't reached even if output is not redirected to a file.

That's correct. Whenever we're in a situation where it's not possible to detect whether output is interactive or not (which is the case for Git for Windows), we should assume it is not, in order to avoid printing ANSI codes to a file. 

> Git for Windows itself has [this detection mechanism](https://github.com/git/git/blob/d4a83d07b8cc66d4afac2f33a8af729f2ba93bba/compat/winansi.c#L579-L585)

Yeah, that was the workaround I mentioned that we could add to `Win32MppImpls` in a separate PR.

Reply:
We might as well merge the envvar part now, since it's still valid for systems where the interactivity is detected correctly.

Reply:
Ok. I've rewritten my commit to not include the changes to the `interactive` logic.

Reply:
> Yeah, that was the workaround I mentioned that we could add to `Win32MppImpls` in a separate PR.

@ajalt are you going to do that?